lib/spack/llnl/util/tty/__init__.py

```diff
@@ -53,10 +59,29 @@ def set_debug(flag):
 def set_verbose(flag):
     global _verbose
     _verbose = flag
+    
+
+def set_stacktrace(flag):
+    global _stacktrace
+    _stacktrace = flag
+
+
+def process_stacktrace(countback):
+    """Gives file and line frame 'countback' frames from the bottom"""
+    st = traceback.extract_stack()
+    # First entry should be bin/spack. Use this to get spack 'root'.
+    # bin/spack is 9 characters, the length of the 'root' is then len-9.
+    root_len = len(st[0][0]) - 9
+    st_idx = len(st) - countback - 1
+    st_text = "%s:%i " % (st[st_idx][0][root_len:], st[st_idx][1])
```

Comment:
I've updated and and rebased the branch, and added a better method for getting the spack root directory.

All files in the stacktrace will be spack files and hence we can just use os.path.commonprefix to fetch the commonprefix.